src/java.base/unix/native/libnio/ch/UnixDomainSockets.c

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright (c) 2001, 2020, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+#include <poll.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <string.h>
+#include <stddef.h>
+#include <netinet/in.h>
+#include <netinet/tcp.h>
+#include <limits.h>
+
+#include "jni.h"
+#include "java_props.h"
+#include "jni_util.h"
+#include "jvm.h"
+#include "jlong.h"
+#include "sun_nio_ch_Net.h"
+#include "nio_util.h"
+#include "nio.h"
+
+/* Subtle platform differences in how unnamed sockets (empty path)
+ * are returned from getsockname()
+ */
+#ifdef MACOSX
+  #define ZERO_PATHLEN(len) (JNI_FALSE)
+#else
+  #define ZERO_PATHLEN(len) (len == offsetof(struct sockaddr_un, sun_path))
+#endif
+
+jbyteArray sockaddrToUnixAddressBytes(JNIEnv *env, struct sockaddr_un *sa, socklen_t len) {
+
+    if (sa->sun_family == AF_UNIX) {
+        int namelen;
+        if (ZERO_PATHLEN(len)) {
+            namelen = 0;
+        } else {
+            namelen = strlen(sa->sun_path);
+        }
+        jbyteArray name = (*env)->NewByteArray(env, namelen);
+        if (namelen != 0) {
+            (*env)->SetByteArrayRegion(env, name, 0, namelen, (jbyte*)sa->sun_path);
+        }
```

Comment:
Should the exception status be checked after calling `SetByteArrayRegion`?

Reply:
I notice a lot of other code is not fussy about checking this, but it would look safer to check and return NULL in this case.